lib/srv/sess.go

```diff
@@ -214,6 +214,7 @@ func (s *SessionRegistry) OpenExecSession(channel ssh.Channel, req *ssh.Request,
 
 	// Start a non-interactive session (TTY attached). Close the session if an error
 	// occurs, otherwise it will be closed by the callee.
+	ctx.session = sess
```

Comment:
Could you add a unit or integration test that would catch this?

Reply:
revised: https://github.com/gravitational/teleport/pull/3955/commits/c334ef15b9aeb422c09467c639311f1d5e578f26